File: .sdk-repos/version-8.9/orchestration-cluster-api-python/examples/role.py
# Compilable usage examples for role management operations.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import (
    CamundaClient,
    MappingRuleSearchQueryRequest,
    RoleCreateRequest,
    RoleGroupSearchQueryRequest,
    RoleSearchQueryRequest,
    RoleUpdateRequest,
    Unset,
    Username,
)


# region CreateRole
def create_role_example() -> None:
    client = CamundaClient()

    result = client.create_role(
        data=RoleCreateRequest(role_id="developer", name="Developer"),
    )

    print(f"Role: {result.role_id}")
# endregion CreateRole


# region GetRole
def get_role_example() -> None:
    client = CamundaClient()

    result = client.get_role(role_id="developer")

    print(f"Role: {result.name}")
# endregion GetRole


# region SearchRoles
def search_roles_example() -> None:
    client = CamundaClient()

    result = client.search_roles(
        data=RoleSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for role in result.items:
            print(f"Role: {role.name}")
# endregion SearchRoles


# region UpdateRole
def update_role_example() -> None:
    client = CamundaClient()

    client.update_role(
        role_id="developer",
        data=RoleUpdateRequest(name="senior-developer"),
    )
# endregion UpdateRole


# region DeleteRole
def delete_role_example() -> None:
    client = CamundaClient()

    client.delete_role(role_id="developer")
# endregion DeleteRole


# region AssignRoleToUser
def assign_role_to_user_example(username: Username) -> None:
    client = CamundaClient()

    client.assign_role_to_user(
        role_id="developer",
        username=username,
    )
# endregion AssignRoleToUser


# region UnassignRoleFromUser
def unassign_role_from_user_example(username: Username) -> None:
    client = CamundaClient()

    client.unassign_role_from_user(
        role_id="developer",
        username=username,
    )
# endregion UnassignRoleFromUser


# region AssignRoleToGroup
def assign_role_to_group_example() -> None:
    client = CamundaClient()

    client.assign_role_to_group(
        role_id="developer",
        group_id="engineering",
    )
# endregion AssignRoleToGroup


# region UnassignRoleFromGroup
def unassign_role_from_group_example() -> None:
    client = CamundaClient()

    client.unassign_role_from_group(
        role_id="developer",
        group_id="engineering",
    )
# endregion UnassignRoleFromGroup


# region AssignRoleToClient
def assign_role_to_client_example() -> None:
    client = CamundaClient()

    client.assign_role_to_client(
        role_id="developer",
        client_id="my-service-account",
    )
# endregion AssignRoleToClient


# region UnassignRoleFromClient
def unassign_role_from_client_example() -> None:
    client = CamundaClient()

    client.unassign_role_from_client(
        role_id="developer",
        client_id="my-service-account",
    )
# endregion UnassignRoleFromClient


# region AssignRoleToMappingRule
def assign_role_to_mapping_rule_example() -> None:
    client = CamundaClient()

    client.assign_role_to_mapping_rule(
        role_id="developer",
        mapping_rule_id="rule-123",
    )
# endregion AssignRoleToMappingRule


# region UnassignRoleFromMappingRule
def unassign_role_from_mapping_rule_example() -> None:
    client = CamundaClient()

    client.unassign_role_from_mapping_rule(
        role_id="developer",
        mapping_rule_id="rule-123",
    )
# endregion UnassignRoleFromMappingRule


# region SearchUsersForRole
def search_users_for_role_example() -> None:
    client = CamundaClient()

    result = client.search_users_for_role(
        role_id="developer",
    )

    if not isinstance(result.items, Unset):
        for user in result.items:
            print(f"User: {user.username}")
# endregion SearchUsersForRole


# region SearchGroupsForRole
def search_groups_for_role_example() -> None:
    client = CamundaClient()

    result = client.search_groups_for_role(
        role_id="developer",
        data=RoleGroupSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for group in result.items:
            print(f"Group: {group.group_id}")
# endregion SearchGroupsForRole


# region SearchClientsForRole
def search_clients_for_role_example() -> None:
    client = CamundaClient()

    result = client.search_clients_for_role(
        role_id="developer",
    )

    if not isinstance(result.items, Unset):
        for c in result.items:
            print(f"Client: {c.client_id}")
# endregion SearchClientsForRole


# region SearchMappingRulesForRole
def search_mapping_rules_for_role_example() -> None:
    client = CamundaClient()

    result = client.search_mapping_rules_for_role(
        role_id="developer",
        data=MappingRuleSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for rule in result.items:
            print(f"Mapping rule: {rule.mapping_rule_id}")
# endregion SearchMappingRulesForRole

File: .sdk-repos/version-8.9/orchestration-cluster-api-csharp/examples/Role.cs
// Compilable usage examples for role management operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class RoleExamples
{
    #region CreateRole
    // <CreateRole>
    public static async Task CreateRoleExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.CreateRoleAsync(new RoleCreateRequest
        {
            Name = "developer",
        });

        Console.WriteLine($"Role key: {result.RoleId}");
    }
    // </CreateRole>
    #endregion CreateRole

    #region GetRole

    // <GetRole>
    public static async Task GetRoleExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.GetRoleAsync("developer");
        Console.WriteLine($"Role: {result.Name}");
    }
    // </GetRole>
    #endregion GetRole

    #region SearchRoles

    // <SearchRoles>
    public static async Task SearchRolesExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchRolesAsync(new RoleSearchQueryRequest());

        foreach (var role in result.Items)
        {
            Console.WriteLine($"Role: {role.Name}");
        }
    }
    // </SearchRoles>
    #endregion SearchRoles

    #region UpdateRole

    // <UpdateRole>
    public static async Task UpdateRoleExample()
    {
        using var client = CamundaClient.Create();

        await client.UpdateRoleAsync("developer", new RoleUpdateRequest
        {
            Name = "senior-developer",
        });
    }
    // </UpdateRole>
    #endregion UpdateRole

    #region DeleteRole

    // <DeleteRole>
    public static async Task DeleteRoleExample()
    {
        using var client = CamundaClient.Create();

        await client.DeleteRoleAsync("developer");
    }
    // </DeleteRole>
    #endregion DeleteRole

    #region AssignRoleToUser

    // <AssignRoleToUser>
    public static async Task AssignRoleToUserExample(Username username)
    {
        using var client = CamundaClient.Create();

        await client.AssignRoleToUserAsync("developer", username);
    }
    // </AssignRoleToUser>
    #endregion AssignRoleToUser

    #region UnassignRoleFromUser

    // <UnassignRoleFromUser>
    public static async Task UnassignRoleFromUserExample(Username username)
    {
        using var client = CamundaClient.Create();

        await client.UnassignRoleFromUserAsync("developer", username);
    }
    // </UnassignRoleFromUser>
    #endregion UnassignRoleFromUser

    #region AssignRoleToGroup

    // <AssignRoleToGroup>
    public static async Task AssignRoleToGroupExample()
    {
        using var client = CamundaClient.Create();

        await client.AssignRoleToGroupAsync("developer", "engineering");
    }
    // </AssignRoleToGroup>
    #endregion AssignRoleToGroup

    #region UnassignRoleFromGroup

    // <UnassignRoleFromGroup>
    public static async Task UnassignRoleFromGroupExample()
    {
        using var client = CamundaClient.Create();

        await client.UnassignRoleFromGroupAsync("developer", "engineering");
    }
    // </UnassignRoleFromGroup>
    #endregion UnassignRoleFromGroup

    #region AssignRoleToClient

    // <AssignRoleToClient>
    public static async Task AssignRoleToClientExample()
    {
        using var client = CamundaClient.Create();

        await client.AssignRoleToClientAsync("developer", "my-service-account");
    }
    // </AssignRoleToClient>
    #endregion AssignRoleToClient

    #region UnassignRoleFromClient

    // <UnassignRoleFromClient>
    public static async Task UnassignRoleFromClientExample()
    {
        using var client = CamundaClient.Create();

        await client.UnassignRoleFromClientAsync("developer", "my-service-account");
    }
    // </UnassignRoleFromClient>
    #endregion UnassignRoleFromClient

    #region AssignRoleToMappingRule

    // <AssignRoleToMappingRule>
    public static async Task AssignRoleToMappingRuleExample()
    {
        using var client = CamundaClient.Create();

        await client.AssignRoleToMappingRuleAsync("developer", "rule-123");
    }
    // </AssignRoleToMappingRule>
    #endregion AssignRoleToMappingRule

    #region UnassignRoleFromMappingRule

    // <UnassignRoleFromMappingRule>
    public static async Task UnassignRoleFromMappingRuleExample()
    {
        using var client = CamundaClient.Create();

        await client.UnassignRoleFromMappingRuleAsync("developer", "rule-123");
    }
    // </UnassignRoleFromMappingRule>
    #endregion UnassignRoleFromMappingRule

    #region SearchUsersForRole

    // <SearchUsersForRole>
    public static async Task SearchUsersForRoleExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchUsersForRoleAsync(
            "developer",
            new SearchUsersForRoleRequest());

        foreach (var user in result.Items)
        {
            Console.WriteLine($"User: {user.Username}");
        }
    }
    // </SearchUsersForRole>
    #endregion SearchUsersForRole

    #region SearchGroupsForRole

    // <SearchGroupsForRole>
    public static async Task SearchGroupsForRoleExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchGroupsForRoleAsync(
            "developer",
            new RoleGroupSearchQueryRequest());

        foreach (var group in result.Items)
        {
            Console.WriteLine($"Group: {group.GroupId}");
        }
    }
    // </SearchGroupsForRole>
    #endregion SearchGroupsForRole

    #region SearchClientsForRole

    // <SearchClientsForRole>
    public static async Task SearchClientsForRoleExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchClientsForRoleAsync(
            "developer",
            new SearchClientsForRoleRequest());

        foreach (var c in result.Items)
        {
            Console.WriteLine($"Client: {c.ClientId}");
        }
    }
    // </SearchClientsForRole>
    #endregion SearchClientsForRole

    #region SearchMappingRulesForRole

    // <SearchMappingRulesForRole>
    public static async Task SearchMappingRulesForRoleExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchMappingRulesForRoleAsync(
            "developer",
            new MappingRuleSearchQueryRequest());

        foreach (var rule in result.Items)
        {
            Console.WriteLine($"Mapping rule: {rule.MappingRuleId}");
        }
    }
    // </SearchMappingRulesForRole>
    #endregion SearchMappingRulesForRole
}
